Kona API does not build on Linux

(Completely new to Kona and K)

In trying to compile libkona.a (make lib), the build fails.

I can get libkona.a to build if I comment out the functions sdf and scd (lines 35 and 36 in src/kapi.c).  Not sure what I break when I do this, but it seems to work. 

Thanks!

-P

running "make lib" gives me this:

In file included from src/kapi.c:4:0:
src/kapi.c: In function 'sdf':
src/kona.h:16:11: warning: return makes integer from pointer without a cast [enabled by default]
 #define R return
           ^
src/kapi.c:35:23: note: in expansion of macro 'R'
 I sdf(I i, I(*f)()) { R NYI; } // XXX
                       ^
src/kapi.c: In function 'scd':
src/kapi.c:36:1: warning: parameter names (without types) in function declaration [enabled by default]
 I scd(I i) { I rclose(i); if(r)R FE; R r;} // XXX
 ^
src/kapi.c:36:30: error: 'r' undeclared (first use in this function)
 I scd(I i) { I rclose(i); if(r)R FE; R r;} // XXX
                              ^
src/kapi.c:36:30: note: each undeclared identifier is reported only once for each function it appears in
src/kapi.c:36:34: error: 'FE' undeclared (first use in this function)
 I scd(I i) { I rclose(i); if(r)R FE; R r;} // XXX
                                  ^
src/kapi.c: In function 'jd':
src/kapi.c:40:35: warning: initialization makes pointer from integer without a cast [enabled by default]
 I jd(I d) { I j; K a = gi(d), r = _jd(a); cd(a); j = Ki(r); cd(r); R j; }
                                   ^
src/kapi.c: In function 'dj':
src/kapi.c:41:35: warning: initialization makes pointer from integer without a cast [enabled by default]
 I dj(I j) { I d; K a = gi(j), r = _jd(a); cd(a); d = Ki(r); cd(r); R d; }
                                   ^
make: *** [src/kapi.o] Error 1

--- a/src/kapi.c
+++ b/src/kapi.c
@@ -3,6 +3,7 @@
 #include <unistd.h>
 #include "kona.h"
 
+extern K _jd(K);
 extern K wd(S s, I n);
 extern K dot(K,K);
 extern K newK(I t, I n);
@@ -33,7 +34,7 @@ K gp(S s) {R gpn_(s,strlen(s));}
 
 I sfn(S s, K(*f)(), I i) { R 0; }
 I sdf(I i, I(*f)()) { R NYI; } // XXX
-I scd(I i) { I rclose(i); if(r)R FE; R r;} // XXX
+I scd(I i) { I r=close(i); if(r)R kerr("file"); R r;} // XXX
 
 K gsk(S s, K k) { return gnk(3, gs(sp(s)), k, 0); }
 

